pkg/miscellaneous/miscellaneous.go

```diff
@@ -61,6 +61,42 @@ func GetLatestFullSnapshotAndDeltaSnapList(store brtypes.SnapStore) (*brtypes.Sn
 	return fullSnapshot, deltaSnapList, nil
 }
 
+// Backup contains a full snapshot and all of its delta snapshots.
+type Backup struct {
+	FullSnapshot      *brtypes.Snapshot
+	DeltaSnapshotList brtypes.SnapList
+}
+
+// GetAllBackups returns all backups (full snapshots and all their delta snapshots).
+func GetAllBackups(store brtypes.SnapStore) ([]Backup, error) {
+	var backups []Backup
+
+	snapList, err := store.List()
+	if err != nil {
+		return nil, err
+	}
+
+	backup := Backup{}
+	for i := len(snapList) - 1; i >= 0; i-- {
+		if snapList[i].IsChunk {
+			continue
+		}
+		if snapList[i].Kind == brtypes.SnapshotKindFull {
+			backup.FullSnapshot = snapList[i]
+			backups = append(backups, backup)
+			backup = Backup{}
+			continue
+		}
+		backup.DeltaSnapshotList = append(backup.DeltaSnapshotList, snapList[i])
+	}
+
+	for _, backup := range backups {
+		sort.Sort(backup.DeltaSnapshotList)
+	}
```

Comment:
The `List` call is expected to return the snapshots in sorted order. So, ideally, this is not required.